### PR TITLE
Position tracking button visibility adjustment

### PR DESCRIPTION
**Pull request by VitorVieiraZ**
Position tracking button in ```MMToolbar.qml``` visibility adjustment with new property in ```MMToolbarButton.qml```:
- ```visibilityMode```, which manages the overall visibility of a toolbar button. If set to ```false```, the button will not be displayed on the toolbar under any circumstances. The default value is ```true```, ensuring standard behavior for the button.

```visibilityMode true``` for Position Tracking button

```visibilityMode false```  for Position Tracking button

### CODE REVIEW

File: app/qml/components/MMToolbar.qml [hunk 1]
```diff
@@ -65,10 +65,19 @@ MMBaseToolbar {
 
   function setupBottomBar() {
     var m = root.model
-    var c = m.count
     var w = buttonView.width
     var button
 
+    for ( var j = m.count - 1 ; j >= 0 ; j-- )
+    {
+      if ( m.get(j).visibilityMode === false )
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:73]: You might need to first check if the `visibilityMode` exists. There are buttons like `MMToolbarLongButton` that does not have this new property.

```suggestion
      if ( m.get(j).visibilityMode && m.get(j).visibilityMode === false )
```

File: app/qml/components/MMToolbar.qml [hunk 2]
```diff
@@ -65,10 +65,19 @@ MMBaseToolbar {
 
   function setupBottomBar() {
     var m = root.model
-    var c = m.count
     var w = buttonView.width
     var button
 
+    for ( var j = m.count - 1 ; j >= 0 ; j-- )
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:71]: While you are here, would it be please possible to rename these one-letter variables to something meaningful to make the code more readable? Something like `m -> buttonModel`, `c -> buttonsCount` and `w -> buttonWidth`. Also please change `var` to `let`

**Review by tomasMizera — COMMENTED** [review]
Looking good, just a few notes

File: app/qml/components/MMToolbar.qml [hunk 3]
```diff
@@ -65,10 +65,19 @@ MMBaseToolbar {
...
+    {
+      if ( m.get(j).visibilityMode === false )
+      {
+          m.remove( j );
+      }
+    }
+
+    var c = m.count
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:79]:
```suggestion
  let filteredbuttons = []

    for ( var j = m.count - 1 ; j >= 0 ; j-- )
    {
      if ( m.get(j).visibilityMode || m.get(j).visibilityMode === true )
      {
          filteredbuttons.append( m.get(j) );
      }
    }

    // c = filteredbuttons.length
    // ... replace m.get(i) with filteredbuttons[i]
```

File: app/qml/components/MMToolbar.qml [hunk 4]
```diff
@@ -64,52 +64,60 @@ MMBaseToolbar {
   Loader { id: buttonMore; sourceComponent: componentMore; visible: false }
 
   function setupBottomBar() {
-    var m = root.model
-    var c = m.count
-    var w = buttonView.width
+    var buttonModel = root.model
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:67]:
```suggestion
    let buttonModel = root.model
```

File: app/qml/components/MMToolbar.qml [hunk 5]
```diff
@@ -64,52 +64,60 @@ MMBaseToolbar {
...
+    var buttonWidth = buttonView.width
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:68]:
```suggestion
    let buttonWidth = buttonView.width
```

File: app/qml/components/MMToolbar.qml [hunk 6]
```diff
@@ -64,52 +64,60 @@ MMBaseToolbar {
...
     var button
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:69]:
```suggestion

```

File: app/qml/components/MMToolbar.qml [hunk 7]
```diff
@@ -64,52 +64,60 @@ MMBaseToolbar {
...
+    var buttonsCount = filteredbuttons.length
+
     // add all buttons (max maxButtonsInToolbar) into toolbar
     visibleButtonModel.clear()
-    if(c <= maxButtonsInToolbar || w >= c*root.minimumToolbarButtonWidth) {
-      for( var i = 0; i < c; i++ ) {
-        button = m.get(i)
+    if(buttonsCount <= maxButtonsInToolbar || buttonWidth >= buttonsCount*root.minimumToolbarButtonWidth) {
+      for( var i = 0; i < buttonsCount; i++ ) {
+        button = filteredbuttons[i]
```
Comment (tomasMizera) [inline app/qml/components/MMToolbar.qml:84]:
```suggestion
        let button = filteredbuttons[i]
```

**Review by tomasMizera — CHANGES_REQUESTED** [review]
It does not work in this situation:

 - Open project with position tracking enabled
 - Position tracking button is there ✔️ 
 - Go back to projects
 - Open project _without_ position tracking
 - Position tracking button is still visible ❌

Screenshot 2024-03-14 at 12 53 22

**Comment by VitorVieiraZ** [comment]
Fixed

**Review by tomasMizera — APPROVED** [review]

**Pull request merged**